tests/outlier_detection/test_median.py

```diff
@@ -0,0 +1,205 @@
+import os
+from pathlib import Path
+
+import numpy as np
+import pytest
+
+from stcal.outlier_detection.median import (
+    DiskAppendableArray,
+    MedianComputer,
+    OnDiskMedian,
+    nanmedian3D,
+)
+
+
+def test_disk_appendable_array(tmpdir):
+
+    slice_shape = (8, 7)
+    dtype = "float32"
+    tempdir = Path(tmpdir) / Path("tmptest")
+    Path.mkdir(tempdir)
+    fname = tempdir / "test.bin"
+
+    arr = DiskAppendableArray(slice_shape, dtype, fname)
+
+    # check temporary file setup
+    assert str(arr._filename).split("/")[-1] in os.listdir(tempdir)  # noqa: SLF001
+    assert len(os.listdir(tempdir)) == 1
+    assert arr.shape == (0, *slice_shape)
+
+    # check cwd contains no files
+    assert all(not Path.is_file(Path(f)) for f in os.listdir(tmpdir))
+
+    # check expected append failures
+    candidate = np.zeros((7, 7), dtype=dtype)
+    with pytest.raises(ValueError, match="shape"):
+        arr.append(candidate)
+    candidate = np.zeros((8, 7), dtype="float64")
+    with pytest.raises(ValueError, match="dtype"):
+        arr.append(candidate)
+
+    # check append and read
+    candidate0 = np.zeros(slice_shape, dtype=dtype)
+    candidate1 = np.full(slice_shape, 2, dtype=dtype)
+    candidate2 = np.full(slice_shape, np.nan, dtype=dtype)
+    for candidate in [candidate0, candidate1, candidate2]:
+        arr.append(candidate)
+
+    arr_in_memory = arr.read()
+
+    assert arr_in_memory.shape == (3, *slice_shape)
+    assert np.all(arr_in_memory[0] == candidate0)
+    assert np.all(arr_in_memory[1] == candidate1)
+    assert np.allclose(arr_in_memory[2], candidate2, equal_nan=True)
+
+
+def test_disk_appendable_array_bad_inputs(tmpdir):
+
+    slice_shape = (8, 7)
+    dtype = "float32"
+    tempdir = Path(tmpdir) / Path("tmptest")
+    fname = "test.bin"
+
+    # test input directory does not exist
+    with pytest.raises(FileNotFoundError):
+        DiskAppendableArray(slice_shape, dtype, tempdir / fname)
+
+    # make the input directory
+    Path.mkdir(tempdir)
+
+    # ensure failure if slice_shape is not 2-D
+    with pytest.raises(ValueError, match="slice shape"):
+        DiskAppendableArray((3, 5, 7), dtype, tempdir / fname)
+
+    # ensure failure if dtype is not valid
+    with pytest.raises(TypeError):
+        DiskAppendableArray(slice_shape, "float3", tempdir / fname)
+
+    # ensure failure if pass directory instead of filename
+    with pytest.raises(IsADirectoryError):
+        DiskAppendableArray(slice_shape, "float3", tempdir)
+
+
+# filter warnings such that "ResourceWarning: Implictly cleaning up" is raised
+# unfortunately need to do this indirectly
+# by finding "finalize" string in PytestUnraisableException
+@pytest.mark.filterwarnings(
+        "error:.*finalize.*:pytest.PytestUnraisableExceptionWarning"
+        )
```

Comment:
```suggestion
```
I don't see this warning if I remove this filter. Is it still needed?

Reply:
The filter ensures that the test fails if cleanup didn't occur properly.  Maybe instead I should make an expected failure related to this.  Will look into doing that

Reply:
Thanks! I'd say remove it then. If the test fails I'd be ok seeing a second failure because of the warning. That seems preferable to ignoring the warning (which we do want to produce an error or at least a warning in case a future update to the code in this PR breaks `cleanup`).

Reply:
We aren't ignoring the warning, we are upgrading it to a failure so that the test does fail if a PR breaks cleanup

Reply:
Thanks. Yeah I certainly got confused there. What about adding `PytestUnraisableExceptionWarning` to `pyproject.toml` instead:
https://github.com/spacetelescope/stcal/blob/4ba7a620ad6545067695fe3808b405a7a50e4d53/pyproject.toml#L102-L104
Hopefully it doesn't cause unrelated failures but if it does I'd be in favor of adding it as a separate PR and leaving the filtering out of this one.

Reply:
It appears not to cause any unrelated failures so I went ahead and added it